includes/class-wc-product-query.php

```diff
@@ -0,0 +1,74 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class for parameter-based Product querying.
+ * Args and usage: https://github.com/woocommerce/woocommerce/wiki/wc_get_products-and-WC_Product_Query
+ *
+ * @version  3.2.0
+ * @since    3.2.0
+ * @package  WooCommerce/Classes
+ * @category Class
+ */
+class WC_Product_Query extends WC_Object_Query {
+
+	/**
+	 * Valid query vars for products.
+	 * @return array
+	 */
+	protected function get_default_query_vars() {
+		return array_merge(
+			parent::get_default_query_vars(),
+			array(
+				'status'             => array( 'draft', 'pending', 'private', 'publish' ),
+				'type'               => array_merge( array_keys( wc_get_product_types() ) ),
```

Comment:
Do you think we should keep separate `downloadable` and `virtual` types/props, or bundle them here? Do we support querying multiple types at once?